Test compatibility with the latest setuptools_scm

It seems that this assert:

https://github.com/ofek/hatch-vcs/blob/425d529cd24295311ca5b52b589a6308165fb244/tests/test_build.py#L81

no longer works with the latest setuptools_scm, because the version file newly contains typing info comment at the end of the line, see: https://github.com/pypa/setuptools_scm/pull/907/files

The exact output from

```
python3 -m venv _e
. _e/bin/activate
pip install -e .
pip install pytest
pytest
```

is

```
[…]
>       assert version_line.endswith(" = '1.2.3'")
E       assert False
E        +  where False = <built-in method endswith of str object at 0x7f7025df4090>(" = '1.2.3'")
E        +    where <built-in method endswith of str object at 0x7f7025df4090> = "__version__ = version = '1.2.3'  # type: str".endswith

/home/ben/src/forks/hatch-vcs/tests/test_build.py:81: AssertionError
```

I think I agree with the suggestion in https://src.fedoraproject.org/rpms/python-hatch-vcs/pull-request/14 to change `assert version_line.endswith(" = '1.2.3'")` to `assert " = '1.2.3'" in version_line`. I’ll open a PR.

--- a/tests/test_build.py
+++ b/tests/test_build.py
@@ -78,7 +78,7 @@ def test_write(new_project_write):
     version_starts = ('version = ', '__version__ = ')
     assert any(line.startswith(version_starts) for line in lines)
     version_line = next(line for line in lines if line.startswith(version_starts))
-    assert version_line.endswith(" = '1.2.3'")
+    assert " = '1.2.3'" in version_line
 
 
 @pytest.mark.skipif(sys.version_info[0] == 2, reason='Depends on fix in 6.4.0 which is Python 3-only')

